File: cap06/python/tablero3enraya.py
# constantes
X = "X"
O = "O"
VACIA = " "
N = "\n"

# tableros de ejemplo
fichasTableroVacio = [
        [VACIA, VACIA, VACIA],
        [VACIA, VACIA, VACIA],
        [VACIA, VACIA, VACIA]
    ]

tableroLleno = [ 
        [VACIA, O, VACIA], 
        [X, O, X], 
        [X, O, VACIA]
    ]

def pintarTablero(tablero):
    texto = ""
    for i in range(len(tablero)):
        # línea con las fichas
        for j in range(len(tablero[i])):
            texto += " " + tablero[i][j] + " "
            if j != len(tablero[i]) - 1:
                texto += "|"
            else:
                texto += N
        # línea de separación
        if i != len(tablero) - 1:
            for j in range(len(tablero[i])):
                texto += "---"
                if j != len(tablero[i]) - 1:
                    texto += "+"
                else:
                    texto += N
        else:
            texto += N
    print(texto)

print("Tablero vacío:")
pintarTablero(fichasTableroVacio)
         
print("Tablero lleno:")
pintarTablero(tableroLleno)

print("Tablero vacío en 1 línea:")
tableroVacio1Linea = \
     ("   |" * 2 + "   \n" + "---+" * 2 + "---\n") * 2 \
    + "   |" * 2 + "   \n"
print(tableroVacio1Linea)

File: cap06/java/Tablero3EnRaya.java
public class Tablero3EnRaya {

	private static final String X = "X";
	private static final String O = "O";
	private static final String VACIA = " ";

	private static final String N = System.lineSeparator();

	// Tableros de ejemplo
	private static final String[][] TABLERO_VACIO = { 
			{ VACIA, VACIA, VACIA }, 
			{ VACIA, VACIA, VACIA },
			{ VACIA, VACIA, VACIA } 
		};
	private static final String[][] TABLERO_LLENO = { 
			{ VACIA, O, VACIA }, 
			{ X, O, X }, 
			{ X, O, VACIA } 
		};

	public static final void pintarTablero(String[][] tablero) {
		String texto = "";
		for (int i = 0; i < tablero.length; i++) {
			for (int j = 0; j < tablero[i].length; j++) {
				texto += VACIA + tablero[i][j] + VACIA;
				if (j != tablero[i].length - 1) {
					texto += "|";
				} else {
					texto += N;
				}
			} // for j
			if (i != tablero.length - 1) {
				for (int j = 0; j < tablero[i].length; j++) {
					texto += "---";
					if (j != tablero[i].length - 1) {
						texto += "+";
					} else {
						texto += N;
					}
				}
			} else {
				texto += N;
			}
		} // for is
		System.out.println(texto);
	}

	public static void main(String[] args) {
		System.out.println("Tablero vacío:");
		pintarTablero(TABLERO_VACIO);
		System.out.println("Tablero lleno:");
		pintarTablero(TABLERO_LLENO);
	}
}
